src/components/pill.tsx

```diff
@@ -1,15 +1,19 @@
 import * as React from 'react';
 import classNames from 'classnames';
 
+const PILL_CLASSNAME = 'pill';
+const PILL_CLASSNAME_ACTIVE = 'pill--active';
+
 export interface PillProps {
-    /** Whether the pill should be shown as active */
     active?: boolean
 }
 
-export const Pill: React.StatelessComponent<PillProps> = ({ active, children }) => {
-    const classes = classNames('pill', { active });
+export const Pill: React.FC<PillProps> = ({ active, children }) => {
+    const classes = classNames(PILL_CLASSNAME, {
+        [PILL_CLASSNAME_ACTIVE]: active,
+    });
 
-    return <div className={classes}>{children}</div>;
+    return <span className={classes}>{children}</span>;
```

Comment:
Why?

Reply:
I think because a pill is usually an  inline element instead of block level element? Probably there's CSS now that overrides the div's block display to inline.

Reply:
If we do this, the children has to be restricted in some form. Otherwise this can produce invalid HTML.

Reply:
Like a `div` a `span` does not mean anything on its own, but user agent CSS makes a `div` a block level element and a `span` an inline element. As far as I could see, the way a `.pill` was used only inline so a `span` seems more appropriate.

Reply:
Ok. I always thought of them more as floating divs, but I guess I don't really care too strongly one way or another.